### PR TITLE
add timeout

### PR DESCRIPTION
**Pull request by Ericgig**
**Description**
Since yesterday, our tests hangs for both `master` and `dev.major` and never finish. It seen only a few tests are affected.
I am adding a timeout so the tests that hangs result in a localized error.

### CODE REVIEW

**Comment by hodgestar** [comment]
The timeouts are because of a GitHub actions outage. It appears to have been resolved earlier today -- https://www.githubstatus.com/incidents/29xnrws671f0.

**Comment by Ericgig** [comment]
This run will still timeout, as it was started before the fix.
Hopefuly it will work on a rerun.

**Comment by Ericgig** [comment]
Still failing... I will retry tomorrow in case it just take time to apply the fix.

**Comment by hodgestar** [comment]
Maybe macos needs a slightly longer timeout? It is a bit slow.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -145,6 +145,8 @@ jobs:
           #  --strict-markers
           #     error out if a marker is used but not defined in the
           #     configuration file
+          #  --timeout=150
```
Comment (hodgestar) [inline .github/workflows/tests.yml:148]:
```suggestion
          #  --timeout=300
```

**Pull request merged**